### PR TITLE
Allocate memory in analysis

### PR DESCRIPTION
**Pull request by tmigot**
Alternative to PR #74 
- allocate memory for the factorization in `ldl_analyze`
~~- set `__factorized` as true if the factorization is successful~~

### CODE REVIEW

**Comment by dpo** [comment]
Seems good to me.

@geoffroyleconte Any comments?

**Comment by geoffroyleconte** [comment]
maybe remove `__factorized` from the struct if it is not needed anymore? Or you plan it to use it in another PR to indicated if the factorization failed @tmigot ?

**Comment by tmigot** [comment]
Yes, you are right. The plan was to keep `__factorized` for the factorization in a second PR.

**Comment by dpo** [comment]
Also `ldl_solve` should really check that `__factorized` is `true`.

**Comment by geoffroyleconte** [comment]
OK nice! Then it's good for me too @dpo

**Comment by dpo** [comment]
> ldl_solve should really check that __factorized is true.

**Comment by tmigot** [comment]
@geoffroyleconte and @dorban
I finally added two things, we discussed:
- Set `__factorized` to `true` if the factorization is successful and `false` otherwise.
- Return an error if trying to solve or multiply with an unsuccessful factorization.

File: test/runtests.jl [hunk 1]
```diff
@@ -280,15 +280,26 @@ end
...
   @test !S.__factorized
   _allocs1 = @allocated ldl_factorize!(M1, S)
   @test S.d[11:20] ≈ -ϵ * ones(10)
+  @test S.__factorized
   _allocs2 = @allocated ldl_factorize!(M2, S)
   @test S.d[11:20] ≈ -100ϵ * ones(10)
   @test _allocs1 == _allocs2
+
+  @test_throws LDLFactorizations.SQDException ldl_factorize!(M3, S)
+  @test !S.__factorized
```
Comment (dpo) [inline test/runtests.jl:298]: It's fine to access this field internally, but we shouldn't encourage access to it from the outside. That's why I prepended its name with `__`. Maybe there could be a `factorized(S) = S.__factorized` method?

**Review by dpo — CHANGES_REQUESTED** [review]
Looking good! Just one request.

File: test/runtests.jl (hunk 1 above)
Comment (tmigot) [reply test/runtests.jl]: Great idea. I added such a function.

**Comment by dpo** [comment]
Great, thank you!

**Pull request merged**